website/docs/r/virtual_machine.html.markdown

```diff
@@ -1077,7 +1079,9 @@ The options available in the `clone` block are:
 
 * `template_uuid` - (Required) The UUID of the source virtual machine or template.
 
-* `linked_clone` - (Optional) Clone the virtual machine from a snapshot or a template. Default: `false`.
+* `instant_clone` - (Optional) Clone the virtual machine from a running powered on virtual machine. Supported on vSphere 6.7 and later. Default: `false`. Only one of `instant_clone` or `linked_clone` can be set to `true`.
+
+* `linked_clone` - (Optional) Clone the virtual machine from a snapshot or a template. Default: `false`. Only one of `linked_clone` or `instant_clone` can be set to `true`.
```

Comment:
The fact that the two clone modes are mutually exclusive is not actually enforced.
By the looks of it if both are set to true the provider will attempt to do an instant clone.

Just a suggestion - it would be great if you can figure out a neat way to validate that the user is not trying to apply an invalid configuration